ctfcli/cli/challenges.py

```diff
@@ -51,27 +51,35 @@ def templates(self):
 
         Templates().list()
 
-    def add(self, repo):
+    def add(self, repo, category=None):
         config = load_config()
 
         if repo.endswith(".git"):
             # Get relative path from project root to current directory
             challenge_path = Path(os.path.relpath(os.getcwd(), get_project_path()))
 
-            # Get new directory that will exist after clone
+            # Get new directory that will add the git subtree
             base_repo_path = Path(os.path.basename(repo).rsplit(".", maxsplit=1)[0])
 
-            # Join targets
-            challenge_path = challenge_path / base_repo_path
+            # Set the challenge path's category directory
+            if category:
+                category_path = Path(category)
+                # Join targets
+                challenge_path = challenge_path / category_path / base_repo_path
+            else:
+                challenge_path = challenge_path / base_repo_path
```

Comment:
It's really unclear what this is trying to do. I think you should have all that you need to know from os.getcwd(), get_project_path(), and base_repo_path.